intuita-webview/src/jobDiffView/App.tsx

```diff
@@ -12,45 +12,54 @@ const getViewComponent = (view: View) => {
 	switch (view.viewId) {
 		case 'jobDiffView':
 			const { data } = view.viewProps;
-			return <JobDiffView {...data} />;
+			return data.map((el) => <JobDiffView key={el.jobHash} {...el} />);
 		default:
 			return null;
 	}
 };
 
 function App() {
 	const [view, setView] = useState<View | null>(null);
-
-	useEffect(() => {
-		const handler = (e: MessageEvent<WebviewMessage>) => {
-			const { data: message } = e;
+	const eventHandler = useCallback(
+		(event: MessageEvent<WebviewMessage>) => {
+			const { data: message } = event;
 			if (message.kind === 'webview.global.setView') {
 				setView(message.value);
 			}
-			if (message.kind === 'webview.diffView.updateDiffViewProps') {
-				const { data } = message;
-				if (
-					view?.viewId === 'jobDiffView' &&
-					view.viewProps.data.jobHash === data.jobHash
-				) {
-					setView({
-						...view,
-						viewProps: {
-							data: {
-								...data,
-							},
-						},
-					});
+			if (
+				message.kind === 'webview.diffView.updateDiffViewProps' &&
+				view?.viewId === 'jobDiffView'
+			) {
+				const jobHash = message?.data?.jobHash;
+				if (!jobHash) {
+					return;
+				}
+				const viewData = [...view?.viewProps?.data];
```

Comment:
`view.viewProps.data.slice() ?? []`
I think that `?` should not be required at the line

Reply:
@henoktsegayetoptal 's part
Benny: addressed it